analyses/mutational-signatures/run_mutational_signatures.sh

```diff
@@ -12,35 +12,12 @@ cd "$(dirname "${BASH_SOURCE[0]}")"
 # In CI we'll run an abbreviated version of the de novo signatures extraction
 ABBREVIATED_MUTSIGS=${OPENPBTA_QUICK_MUTSIGS:-0}
 
-data_dir=../../data
-denovo_plot_dir=plots/denovo
-
 # Run the mutational signatures analysis using existing signatures
 Rscript -e "rmarkdown::render('01-known_signatures.Rmd', clean = TRUE)"
 
-# if abbreviated mutsigs is "true", we'll run only one number of signatures and
-# for an unacceptably low number of iterations to do any analysis with
-if [ "$ABBREVIATED_MUTSIGS" -gt "0" ]
-then
-  FLOOR=10
-  CEILING=10
-  NUM_ITER=10
-else
-  FLOOR=5
-  CEILING=15
-  NUM_ITER=1000
-fi
-
-# Directory to hold the goodness-of-fit plot
-mkdir -p $denovo_plot_dir
+# Split up the consensus MAF files by experimental strategy (writes to scratch)
+Rscript --vanilla 02-split_experimental_strategy.R
 
-# De novo signatures extraction
-Rscript --vanilla \
-  scripts/de_novo_signature_extraction.R \
-  --maf_file "${data_dir}/pbta-snv-consensus-mutation.maf.tsv.gz" \
-  --nsignatures_floor $FLOOR \
-  --nsignatures_ceiling $CEILING \
-  --num_iterations $NUM_ITER \
-  --seed 42 \
-  --output_file "results/denovo_sigfit_signatures.RDS" \
-  --plot_output "${denovo_plot_dir}/denovo_sigfit_${FLOOR}_to_${CEILING}_gof.pdf"
+# Run the shell script that is for determining the number of signatures to use
+# with a low number of iterations
+QUICK_MUTSIGS=$ABBREVIATED_MUTSIGS bash 03-de_novo_range_of_nsignatures.sh
```

Comment:
I guess my question here is why do this step in a separate bash script rather than just calling de_novo_signature_extraction.R  as part of this one? I don't have a strong opinion, but I am not quite sure of the modularity advantage here, weighed against the indirectness. 

Phrased differently, why does `de_novo_signature_extraction.R` have so many required options that have to be set at the command line, while `02-split_experimental_strategy.R` uses default write locations, etc.?

Reply:
There is nothing specific about `de_novo_signature_extraction.R` to this project, in my opinion. (All the column names used are standard MAF stuff as far as I know.) Having to join the `pbta-histologies.tsv` file to the `Tumor_Sample_Barcodes` and using `WXS` to refer to whole exome sequencing - these all seem pretty specific to the project to me. I could make splitting into experimental strategies step more general but decided the development time cost outweighed the benefits. I made the opposite decision for `de_novo_signature_extraction.R` because I thought it had more potential for future use and because we will need to run that at least one more time in the context of this module with different parameters (probably more than that with different seeds...). 

My reasoning for the move to numbering a bash script is almost certainly unclear from this PR, because it's future-looking in a way (and I didn't say anything about why when filing 😄 ).

In the absence of moving the step where we are looking at a range of values for `nsignatures` to its own numbered script, the module structure would probably will have been something like:

```
01 look at known signatures
02 split by strategy
03 pick the number of signatures
04 match to known signatures
05 bubble plot, etc.
run_mutational_signatures.sh
```

Where between `02` and `03` we'd run the range of `nsignatures` values and between `03` and `04` we'd run signature extraction with a set number of signatures and a high number of iterations.

I thought moving to a separate shell script would be more clear from that view than having those steps in `run_mutational_signatures.sh`. On the other hand, there will probably be a step in `run_mutational_signatures.sh` that runs `de_novo_signature_extraction.R` with the selected number of signatures for a high number of iterations. Based on discussion on #799, though, I think we might need to do that with multiple values for the `seed` argument which might be a rationale for moving it into it's own script, too. 

That was a long winded way to say that the own bash script decision is not set in stone and I feel more strongly about the design decisions I made with the R scripts given the information I have.